src/view/Office.vue

```diff
@@ -111,7 +178,37 @@ export default {
 				PostMessages.sendWOPIPostMessage(FRAME_DOCUMENT, msgId, values)
 			},
 		})
-		PostMessages.registerPostMessageHandler(({ parsed }) => {
+		PostMessages.registerPostMessageHandler(this.postMessageHandler)
+		this.load()
+	},
+	beforeDestroy() {
+		PostMessages.unregisterPostMessageHandler(this.postMessageHandler)
+	},
+	methods: {
+		async load() {
+			const isPublic = document.getElementById('isPublic') && document.getElementById('isPublic').value === '1'
+			this.src = getDocumentUrlForFile(this.filename, this.fileid) + '&path=' + encodeURIComponent(this.filename)
+			if (isPublic) {
+				this.src = getDocumentUrlForPublicFile(this.filename, this.fileid)
+			}
+			this.loading = LOADING_STATE.LOADING
+			this.loadingTimeout = setTimeout(() => {
+				console.error('FAILED')
+				this.loading = LOADING_STATE.FAILED
+				this.error = t('richdocuments', 'Failed to load {productName} - please try again later', { productName: loadState('richdocuments', 'productName', 'Nextcloud Office') })
+			}, (OC.getCapabilities().richdocuments.config.timeout * 1000 || 15000))
+		},
+		documentReady() {
+			this.loading = LOADING_STATE.DOCUMENT_READY
+			clearTimeout(this.loadingTimeout)
+		},
+		async share() {
+			FilesAppIntegration.share()
+		},
+		close() {
+			this.$parent.close()
+		},
```

Comment:
Couldn't this be done with an event to make it more robust?

Reply:
Would need support in the viewer first, but might be a good idea to introduce that there.